src/cmake-action.ts

```diff
@@ -26,6 +28,12 @@ export async function main(): Promise<void> {
     const testPresetAdditionalArgs = actionLib.getInput(cmakeglobals.testPresetAdditionalArgs, false);
     const packagePresetAdditionalArgs = actionLib.getInput(cmakeglobals.packagePresetAdditionalArgs, false);
     const runVcpkgEnvFormatString = actionLib.getInput(vcpkgglobals.runVcpkgEnvFormatStringInput, false);
+
+    // Set parallelization level
+    const nproc = os.availableParallelism();
```

Comment:
From what I've read `os.availableParallelism` should take into account the virtualization and get the actual available processors, i.e. `nproc` vs `nproc --all`